backend/src/apiserver/server/pipeline_server.go

```diff
@@ -191,11 +191,14 @@ func (s *PipelineServer) CreatePipelineV1(ctx context.Context, request *apiv1bet
 	}
 
 	// A new pipeline has been created, but pipeline version creation failed after validation
+	// This block should not be executed when an existing pipeline is found
 	if perr == nil {
```

Comment:
Assuming this logic will be move to v2 method, this check should be `if perr != nil`, right?

Reply:
In this case, `perr` is `nil` only when a new pipeline is created - no existing pipelines with such (name, namespace)